api/v1alpha1/varnishcluster_types.go

```diff
@@ -95,7 +100,18 @@ type VarnishClusterSpec struct {
 	// +kubebuilder:validation:Enum=debug;info;warn;error;dpanic;panic;fatal
 	LogLevel string `json:"logLevel,omitempty"`
 	// +kubebuilder:validation:Enum=json;console
-	LogFormat string `json:"logFormat,omitempty"`
+	LogFormat      string          `json:"logFormat,omitempty"`
+	HaproxySidecar *HaproxySidecar `json:"haproxySidecar,omitempty"`
+}
+
+type HaproxySidecar struct {
+	Enabled       bool   `json:"enabled,omitempty"`
+	ConfigMapName string `json:"configMapName,omitempty"` // mount under /usr/local/etc/haproxy/haproxy.cfg
+	Image         string `json:"image,omitempty"`
+	// +kubebuilder:validation:Enum=Always;Never;IfNotPresent
+	ImagePullPolicy v1.PullPolicy            `json:"imagePullPolicy,omitempty"`
+	ImagePullSecret string                   `json:"imagePullSecret,omitempty"`
+	Resources       *v1.ResourceRequirements `json:"resources,omitempty"`
```

Comment:
Should be safe to have it as not a pointer. That way you won't need to deal with nil checks and dereferences. Kubernetes also accepts value, not a pointer.